### ISSUE TITLE
Sentry inconsistent capturing messages and exceptions - node

### ISSUE THREAD

**Issue by kinluek** [getsentry/sentry#10911]
For some reason sentry is being really inconsistent with capturing messages and exceptions, sometimes they show sometimes they don't, but almost every time an exception is thrown it screws with things showing up. Generally it is fine when just capturing messages but not all the time 

I am using the latest Sentry SDK on node. I am running a program that fires multiple async functions within a promise.all and I am trying to capture messages and exceptions within these functions, I am wondering if there are any weird quirks when using sentry across multiple concurrently running async functions.

Thanks

**Comment by markstory** [comment]
How can someone reproduce the inconsistencies you're seeing?

**Comment by kinluek** [comment]
So I went on to build that dummy project but couldn't reproduce the inconsistencies, what I ended up doing was replacing all the await functions with awaiting promisified setTimeouts keeping the general program layout in place and removing all of the complex code.

After a long time of playing around I went back to the original code and basically stuck this wait function in the promise.all so that the promise.all as a whole had longer to resolve. This solved my issues and captured all the exceptions and messages I was throwing across all the platformCheckers.runChecks.

heres the little bit of code that fixed it:

```javascript

function wait() {
    return new Promise(resolve => {
        return setTimeout(() => {
            resolve('waited');
        }, 3000);
    });
}

let results = await Promise.all([
            ...platformCheckers.map(async platformChecker => {
                await platformChecker.init();
                return await platformChecker.runChecks();
            }),
            wait()
        ]);

```

This does seem like a bit of a bodge job though...
